Automattic/jetpack
# Connection UI: Building the Framework

#### Changes proposed in this Pull Request:
- The new package
- JS building configuration
- Handy yarn commands
- The WP admin menu link

The new commands (run from `packages/connection-ui/` directory):
- `yarn build`
- `yarn watch`

The Jetpack command `yarn build-packages` was also edited to also build the Connection UI package.

#### Jetpack product discussion
p9dueE-1vM-p2
Related to #18182

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Go to `packages/connection-ui` and run `yarn install` to install the required modules.
2. Run `yarn build`, confirm no errors come up.
3. Go to WP admin area, "Tools -> Connection Manager", you should see the "Connection Manager" header. That means the build was successful.
4. Run `yarn watch`.
5. Open the file `packages/connection-ui/_inc/admin.jsx` and change the `Connection Manager` text to something else.
6. Wait a few seconds, reload the "Tools -> Connection Manager" page, make sure the changed text appears.
7. Stop the watching process (Ctrl+C).
8. Revert the file `packages/connection-ui/_inc/admin.jsx` to its original state.
9. Go to the Jetpack root directory and run `yarn build-packages`. Confirm that it shows no errors.
10. Once finished, reload the "Tools -> Connection Manager" page again. Make sure it reflects the latest changes.

_The PR uses a deprecated function `registerStore()` of the `@wordpress/data` component.
Unfortunately, we can't replace it with `createReduxStore()` and `register()` right now, because they are not included into the latest Wordpress build of `@wordpress/data`._

#### Proposed changelog entry for your changes:
n/a
